source/transactions/tests/unified/retryable-abort-handshake.yml

```diff
@@ -0,0 +1,88 @@
+description: "retryable abortTransaction on handshake errors"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "4.2"
+    topologies: [replicaset, sharded, load-balanced]
+    serverless: "forbid"
+    auth: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent, connectionCheckOutStartedEvent]
+      uriOptions: { retryReads: true, retryWrites: true, maxPoolSize: 5 }
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &databaseName retryable-handshake-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collectionName coll
+  - session:
+      id: &session0 session0
+      client: *client0
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+
+tests:
+  - description: "AbortTransaction succeeds after handshake network error"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
```

Comment:
I feel like I'm definitely misunderstanding something, but why does this use `times: 2` like the retryable reads spec's `handshakeError.yml` test? Is that necessary to apply to _both_ the `insert` and `abortTransaction` commands? I'm probably still confused as to why two sequential commands would necessitate checking out separate connections from the pool.

Reply:
Maybe I'm also misunderstanding something 🤔  I definitely think I have something wrong here with the failPoint, I want drivers to be able to successfully checkout a connection and complete the startTransaction and insertOne operation, but abortTransaction needs to have a "network error" (caused by the closeConnection: true setting) one time. 
So I've updated the test to set the failPoint after the insertOne now. I unfortunately cannot run these tests with the node driver due to a fundamental difference in how we connect a client to MongoDB, so I'm doing my best to logic these out and asking Shane to run them against python.